code/modules/surgery/organs/vocal_cords.dm

```diff
@@ -120,3 +85,38 @@
 /obj/item/organ/internal/vocal_cords/colossus/speak_with(message)
 	var/cooldown = voice_of_god(uppertext(message), owner, spans, base_multiplier)
 	next_command = world.time + (cooldown * cooldown_mod)
+
+/obj/item/organ/internal/adamantine_resonator
+	visual = FALSE
+	name = "adamantine resonator"
+	desc = "Fragments of adamantine exist in all golems, stemming from their origins as purely magical constructs. These are used to \"hear\" messages from their leaders."
+	zone = BODY_ZONE_HEAD
+	slot = ORGAN_SLOT_ADAMANTINE_RESONATOR
+	icon_state = "adamantine_resonator"
+
+/obj/item/organ/internal/vocal_cords/adamantine
+	name = "adamantine vocal cords"
+	desc = "When adamantine resonates, it causes all nearby pieces of adamantine to resonate as well. Golems containing these formations use this to broadcast messages to nearby golems."
+	actions_types = list(/datum/action/item_action/organ_action/use/adamantine_vocal_cords)
+	icon_state = "adamantine_cords"
+
+/datum/action/item_action/organ_action/use/adamantine_vocal_cords/Trigger(trigger_flags)
+	if(!IsAvailable(feedback = TRUE))
+		return
+	var/message = tgui_input_text(owner, "Resonate a message to all nearby golems", "Resonate")
+	if(!message)
+		return
+	if(QDELETED(src) || QDELETED(owner))
+		return
+	owner.say(".x[message]")
+
+/obj/item/organ/internal/vocal_cords/adamantine/handle_speech(message)
+	var/msg = span_resonate(span_name("[owner.real_name] resonates, \"[message]\""))
```

Comment:
```suggestion
	var/msg = span_resonate([span_name("[owner.real_name]")] resonates, \"[message]\""))
```

you changed the meaning of the spans such that `span_name` would apply to the entire string, not just the actual name portion. it's not your fault because the last one was so god damn confusing but i'm 99% sure this is what was meant